db/blob/blob_source.cc

```diff
@@ -99,9 +99,11 @@ Status BlobSource::GetBlob(const ReadOptions& read_options,
     Slice key = cache_key.AsSlice();
     s = GetBlobFromCache(key, &blob_entry);
     if (s.ok() && blob_entry.GetValue()) {
-      assert(blob_entry.GetValue()->size() == value_size);
+      // DO NOT REPLACE blob entry size with value_size
+      // The cache entry size might not match value size because value size can
+      // include the on-disk (possibly compressed) size.
       if (bytes_read) {
-        *bytes_read = value_size;
+        *bytes_read = blob_entry.GetValue()->size();
```

Comment:
I'm wondering if it would make sense to change the semantics of `bytes_read` for consistency so it's just the uncompressed size of the value in all cases, even if we read it from the file? This could be handled on the `BlobSource` level, that is, we wouldn't even have to pass the `bytes_read` pointer to `BlobFileReader`

Reply:
Cool! I should probably update how I use the tickers.

Reply:
If it's not trivial, we could also put up a separate 2-liner PR that simply removes the assertions to unblock the crash tests, then get back to this change+compression-related unit tests.

Reply:
Shall we keep the bytes_read for BlobFileReader, many tests still use it. Instead, we can pass in a nullptr from BlobSource level to BlobFileReader if we don't need that information.

Reply:
Sure. We can keep `bytes_read` as-is in `BlobFileReader`.